srw_detector.py

```diff
@@ -66,10 +71,11 @@ def trigger(self):
         sim_id = self._sim_id
         sb = SirepoBluesky(self._sirepo_server)
         data = sb.auth('srw', sim_id)
-        aperture = sb.find_element(data['models']['beamline'], 'title', 'Aperture')
-        aperture['horizontalSize'] = x * 1000
-        aperture['verticalSize'] = y * 1000
-        watch = sb.find_element(data['models']['beamline'], 'title', 'Watchpoint')
+        element = sb.find_element(data['models']['beamline'], 'title', self.cname)
+        element[self.spec_name1] = x * 1000
+        element[self.spec_name2] = y * 1000
```

Comment:
Multiplication by 1000 may not always be the case. You would need to read the units from Sirepo schema and convert the values to those units. I just learned about the [unyt](https://github.com/yt-project/unyt) library, you can try to experiment with that.

Reply:
Going to check this out - knew this was an issue just wasn't quite sure how to approach it. 

Thanks!

Reply:
Please check https://github.com/radiasoft/sirepo/blob/master/sirepo/package_data/static/json/srw-schema.json - that's the schema which contains units in the square brackets for the fields of optics.